### PR TITLE
FIX-2485 Query data grid

### LINKED ISSUE #2485: 💡Improve usability for query view
We want to improve the ease of use of the query view. A feature where you can easily toggle on and off objects you want your query to contain, you can view the checkbox-view or the finished query you are making by checking the checkboxes.
image

Here we would also like the objects to contain ALL sub-objects and not only the most used once.

We would also like to toggle on a feature where you can see what a certain object should contain as this;
The documentation under is based on the WITSML standard
image

### PR DESCRIPTION
**Pull request by eliasbruvik**
[//]: # (Request Template Witsml Explorer)

[//]: # (Thank you for contributing to WITSML Explorer! Before submitting this PR, please fill in the following template.)

## Fixes

[//]: # (Write the GitHub issue number starting with #, or, for Equinor only, the Jira issue number starting with WE-)

This pull request fixes #2485 

## Description

[//]: # (Please include a written summary of the changes and which issue is fixed. List any dependencies that are required for this change._)

Added a "Data Grid" toggle that changes the view between the query editor and a data grid view.
The data grid view shows the value in an editable textfield, and the documentation for the value (as specified in the xsd documents).

Any row that is selected (as marked by a checkbox and a fully marked row) is included in the query.

All objects we normally use in Witsml Explorer is supported in addition to a few extra. However, not all objects are supported yet. These can be added at a later point if requested.

image

## Type of change

[//]: # (Mark any of the types of change that apply.)

* [ ] Bugfix
* [x] New feature (non-breaking change which adds functionality)
* [x] Enhancement of existing functionality
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] This change requires a documentation update

## Impacted Areas in Application

[//]: # (List general components of the application that this PR will affect)

* [x] Frontend
* [ ] API
* [ ] WITSML
* [ ] Desktop
* [ ] Other (please describe)

## Checklist:

[//]: # (Please tick all the boxes or remove the ones that aren't needed and explain why)

*Communication*
* [ ] I have made corresponding changes to the documentation
* [ ] PR affects application security

*Code quality*
* [x] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [ ] New code is covered by passing tests

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by robertbasti** [comment]
When clicking on Pick template and choosing drillReport - it shows "Unable to parse query".  Could not be a better message?

File: Src/WitsmlExplorer.Frontend/components/ContentViews/QueryDataGrid.tsx [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+        checkableRows
+        onRowSelectionChange={onRowSelectionChange}
+        onExpandedChange={setExpanded}
+        rowSelection={rowSelection}
+        expanded={expanded}
+        stickyLeftColumns={2}
+      />
+    </div>
+  ) : (
+    <Typography>Unable to parse query</Typography>
```
Comment (robertbasti) [inline Src/WitsmlExplorer.Frontend/components/ContentViews/QueryDataGrid.tsx]: cound not be there a more descriptive message?

Comment (robertbasti) [reply Src/WitsmlExplorer.Frontend/components/ContentViews/QueryDataGrid.tsx]: e.g. This object is not supported yet.

**Comment by robertbasti** [comment]
when removing e.g. xmlns or version and putting back for example for attachment or other templates, it returns 
Error while querying store: -468 - A queryIn template must include a version attribute in the plural data-object that defines the Data Schema Version of the data-object. It removes the value. Is that desirable behavior?

**Review by robertbasti — DISMISSED** [review]
code seems ok

**Comment by eliasbruvik** [comment]
> When clicking on Pick template and choosing drillReport - it shows "Unable to parse query". Could not be a better message?

Updated the warning.

> when removing e.g. xmlns or version and putting back for example for attachment or other templates, it returns
Error while querying store: -468 - A queryIn template must include a version attribute in the plural data-object that defines the Data Schema Version of the data-object. It removes the value. Is that desirable behavior?

I guess this is intentional (at least for now). If you remove a value from the dataGrid it's also removed in the query. So to get it back you would have to re-specify the values.

**Review by robertbasti — APPROVED** [review]
tested and seems ok

**Pull request merged**